build: serve e2e-app in aot mode

DevVersion opened the pull request:
Starts building the e2e-app in AOT mode

This should make our e2e tests on  the CI faster and also should be a more advanced testing of AOT compatibility.

**Note**: As this also makes the e2e-app smaller in payload I will try to find a way to use a RxJS bundle (would speed-up the serving a lot)

jelbourn commented on src/e2e-app/main-aot.ts:
```diff
@@ -0,0 +1,4 @@
+import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
+import {E2eAppModuleNgFactory} from './e2e-app-module.ngfactory';
+
+platformBrowserDynamic().bootstrapModuleFactory(E2eAppModuleNgFactory);
```
Shouldn't this just be `platformBrowser` (without the `Dynamic`)?

(same for demo-app, now that I notice it)

DevVersion replied:
Yeah I think @josephperrott also had an issue about it but I didn't assign it to myself so I lost track of it. Changed it now.

jelbourn commented on src/e2e-app/main-aot.ts:
```diff
@@ -0,0 +1,4 @@
+import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
```
Instead of having a separate `main-aot.ts`, we should just make this the main `main.ts`; nobody should need to ever run this in JIT mode

DevVersion replied:
I agree. Done!

mmalerba commented:
please rebase

DevVersion commented:
@mmalerba Done. This also includes the fix for the CI failure.

> CI failure appears because we renamed the `constants.ts` file but another PR, that got merged afterwards,  introduced a new file that still imports form `constants.ts`.
